lib/folio_client/source_storage.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class FolioClient
+  # Lookup records in Folio Source Storage
+  class SourceStorage
+    class NotFound < StandardError; end
```

Comment:
Can you use existing error classes? https://github.com/sul-dlss/folio_client/blob/main/lib/folio_client/unexpected_response.rb#L10

Reply:
I'm wondering if we don't want to move those exceptions into a different location. I think they're real easy to miss in their current location.

Reply:
ok, fixed, hopefully.  i also moved the `MultipleRecordsForIdentifier` error class to `UnexpectedResponse`, and renamed it to `MultipleResourcesFound` so it's named a bit more consistently with its `ResourceNotFound` counterpart.  and tweaked the `ResourceNotFound` comment since at least sometimes we get 0 results and a 200 OK instead of a 404 Not Found.  diff between what you reviewed and what i just pushed is:

```diff
diff --git a/lib/folio_client/source_storage.rb b/lib/folio_client/source_storage.rb
index 2a939af..605ef25 100644
--- a/lib/folio_client/source_storage.rb
+++ b/lib/folio_client/source_storage.rb
@@ -3,10 +3,6 @@
 class FolioClient
   # Lookup records in Folio Source Storage
   class SourceStorage
-    class NotFound < StandardError; end
-
-    class MultipleRecordsForIdentifier < StandardError; end
-
     attr_accessor :client
 
     # @param client [FolioClient] the configured client
@@ -21,8 +17,8 @@ class FolioClient
       response_hash = client.get("/source-storage/source-records", {instanceHrid: instance_hrid})
 
       record_count = response_hash["totalRecords"]
-      raise NotFound, "No records found for #{instance_hrid}" if record_count.zero?
-      raise MultipleRecordsForIdentifier, "Expected 1 record for #{instance_hrid}, but found #{record_count}" if record_count > 1
+      raise FolioClient::UnexpectedResponse::ResourceNotFound, "No records found for #{instance_hrid}" if record_count.zero?
+      raise FolioClient::UnexpectedResponse::MultipleResourcesFound, "Expected 1 record for #{instance_hrid}, but found #{record_count}" if record_count > 1
 
       response_hash["sourceRecords"].first["parsedRecord"]["content"]
     end
diff --git a/lib/folio_client/unexpected_response.rb b/lib/folio_client/unexpected_response.rb
index 733d55c..93a1c81 100644
--- a/lib/folio_client/unexpected_response.rb
+++ b/lib/folio_client/unexpected_response.rb
@@ -6,9 +6,12 @@ class FolioClient
     # Error raised by the Folio Auth API returns a 422 Unauthorized
     class UnauthorizedError < StandardError; end
 
-    # Error raised when the Folio API returns a 404 NotFound
+    # Error raised when the Folio API returns a 404 NotFound, or returns 0 results when one was expected
     class ResourceNotFound < StandardError; end
 
+    # Error raised when e.g. exactly one result was expected, but more than one was returned
+    class MultipleResourcesFound < StandardError; end
+
     # Error raised when the Folio API returns a 403 Forbidden
     class ForbiddenError < StandardError; end
 
diff --git a/spec/folio_client/source_storage_spec.rb b/spec/folio_client/source_storage_spec.rb
index 6476cb3..5cfac8d 100644
--- a/spec/folio_client/source_storage_spec.rb
+++ b/spec/folio_client/source_storage_spec.rb
@@ -122,7 +122,7 @@ RSpec.describe FolioClient::SourceStorage do
     }
 
     it "raises a NotFound exception" do
-      expect { source_storage.fetch_marc_hash(instance_hrid:) }.to raise_error(FolioClient::SourceStorage::NotFound, "No records found for #{instance_hrid}")
+      expect { source_storage.fetch_marc_hash(instance_hrid:) }.to raise_error(FolioClient::UnexpectedResponse::ResourceNotFound, "No records found for #{instance_hrid}")
     end
   end
 
@@ -200,7 +200,7 @@ RSpec.describe FolioClient::SourceStorage do
     }
 
     it "raises a MultipleRecordsForIdentifier exception" do
-      expect { source_storage.fetch_marc_hash(instance_hrid:) }.to raise_error(FolioClient::SourceStorage::MultipleRecordsForIdentifier, "Expected 1 record for #{instance_hrid}, but found 2")
+      expect { source_storage.fetch_marc_hash(instance_hrid:) }.to raise_error(FolioClient::UnexpectedResponse::MultipleResourcesFound, "Expected 1 record for #{instance_hrid}, but found 2")
     end
   end
 end
```

Reply:
> I'm wondering if we don't want to move those exceptions into a different location. I think they're real easy to miss in their current location.

ah yeah, i don't know... i had seen them when poking around the gem, but just didn't think to re-use them because... well i'm not actually sure why.  i can't think of a better home off-hand, but definitely open to ideas for moving.